DigitalLearningSolutions.Data/Services/ActionPlanService.cs

```diff
@@ -1,19 +1,31 @@
 ﻿namespace DigitalLearningSolutions.Data.Services
 {
+    using System.Collections.Generic;
     using System.Linq;
+    using System.Threading.Tasks;
     using System.Transactions;
+    using DigitalLearningSolutions.Data.ApiClients;
     using DigitalLearningSolutions.Data.DataServices;
     using DigitalLearningSolutions.Data.DataServices.SelfAssessmentDataService;
+    using DigitalLearningSolutions.Data.Helpers;
+    using DigitalLearningSolutions.Data.Models.LearningResources;
+    using Microsoft.Extensions.Configuration;
 
     public interface IActionPlanService
     {
         void AddResourceToActionPlan(int competencyLearningResourceId, int delegateId, int selfAssessmentId);
+
+        Task<IEnumerable<ActionPlanItem>> GetIncompleteActionPlanItems(int delegateId);
+
+        Task<string?> AccessLearningResource(int learningLogItemId, int delegateId);

```

Comment:
Couldn't really think of a good name for this. Thoughts on this? Might be worth a refactor?

Reply:
I like how short and sweet this name is but it does seem a bit ambiguous. The only other thing I can think of is GetLearningResourceLinkAndUpdateLastAccessedDate which is a mouthful but might be clearer.

Reply:
I like how short and sweet it is but it does seem a bit ambiguous. The only other thing I can think of is GetLearningResourceLinkAndUpdateLastAccessedDate, which is a mouthful but would be clearer